File: 프로그래머스/lv2/42842. 카펫/카펫.swift
import Foundation

func solution(_ brown:Int, _ yellow:Int) -> [Int] {
    var ans = [Int]()
    var f_yellow = Float(yellow)
    for i in 1...Int(sqrt(f_yellow)){
        if(yellow % i == 0){
            var x = yellow / i
            var y = i
            if((x+2)*2 + 2*y == brown){
                ans.append(x+2)
                ans.append(y+2)
                break
            }
        }
    }
    return ans
}

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int leng = brown+yellow;
        for(int i=3;i<leng/2;i++){
            if(leng%i==0){
                if(i>leng/i){
                    break;
                }
                else{
                    if((leng/i)*2+i*2-4 ==brown){
                        answer[0] = leng / i;
                        answer[1] = i;

                    }
                }
            }
        }
        return answer;
    }
}